### PR TITLE
kv/RocksDBStore: using default_cf rather than nullptr.

### PR DESCRIPTION
**Pull request by majianpeng**
Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by majianpeng** [comment]
@tchaikov . please view. thanks!

File: src/kv/RocksDBStore.cc [hunk 1]
```diff
@@ -1378,7 +1378,7 @@ void RocksDBStore::RocksDBTransactionImpl::set(
   } else {
     string key;
     combine_strings(prefix, k, keylen, &key);
-    put_bat(bat, cf, key, to_set_bl);
+    put_bat(bat, db->default_cf, key, to_set_bl);
```
Comment (tchaikov) [inline src/kv/RocksDBStore.cc:1381]: i think it's fine passing a `nullptr` here, as the rockdb uses the nullptr `ColumnFamilyHandle` as a special case. see https://github.com/facebook/rocksdb/blob/722814e357104c3e428b31740b4627239924b973/include/rocksdb/write_batch.h#L78-L82 . before introducing column family support, we were using `Put(const SliceParts& key, const SliceParts& value)`.
after that change, we only use the lower level variant API which accepts `cf` . but it's practically optional.

so as a cleanup, probably you could `s/cf/nullptr/` to make explicit? and note in the commit message that it's a cleanup for better readability?

**Comment by majianpeng** [comment]
@tchaikov . thanks your detail explaination. update by your suggestion. please review.

**Comment by tchaikov** [comment]
> @tchaikov . thanks your detail explaination. update by your suggestion. please review.

no, i still suggest use `nullptr`. it saves a call to `GetColumnFamilyID()` .

and could you fix the typo in the title of your commit message, s/rahter/rather/

**Comment by majianpeng** [comment]
@tchaikov . i think we should not depend on the rocksdb implement. For example, https://github.com/facebook/rocksdb/blob/722814e357104c3e428b31740b4627239924b973/db/db_impl/db_impl.cc#L3314 don't check cf whether nullptr.

**Comment by tchaikov** [comment]
> @tchaikov . i think we should not depend on the rocksdb implement. For example, https://github.com/facebook/rocksdb/blob/722814e357104c3e428b31740b4627239924b973/db/db_impl/db_impl.cc#L3314 don't check cf whether nullptr.

how is this related to the `Put()` call ? could you help me understand the call path ?

**Comment by majianpeng** [comment]
I mean those code in rocksdb inner.  We don't depend on their implementation. Unless doc of API tell cf is nullptr, it mean default_cf.

**Comment by tchaikov** [comment]
> I mean those code in rocksdb inner. We don't depend on their implementation. Unless doc of API tell cf is nullptr, it mean default_cf.

i am not in favor of this change if it's not a pure win. as this change is sitting in the critical path. i don't want to trade this for better encapsulation. also, please note, the "inner" is in the header file or RocksDB, which renders it part of its ABI.

**Comment by majianpeng** [comment]
Except rocksdb, there is leveldb.  So In order to make the code more readable and more compatible, i think it should change.

**Comment by majianpeng** [comment]
@tchaikov .  i'll close this PR.  Thanks!

**Pull request closed without merging**